packages/tests/server/react/useQuery.test.tsx

```diff
@@ -194,6 +205,88 @@ describe('useQuery()', () => {
     expectation.toMatchTypeOf<{ data: '__result' | undefined }>();
     expectation.not.toMatchTypeOf<{ data: undefined }>();
   });
+
+  test('iterable', async () => {
+    const { client, App } = ctx;
+    const states: {
+      status: string;
+      data: unknown;
+    }[] = [];
+    function MyComponent() {
+      const query1 = client.post.iterable.useQuery(undefined, {
+        trpc: {
+          context: {
+            stream: 1,
+          },
+        },
+      });
+      states.push({
+        status: query1.status,
+        data: query1.data,
+      });
+
+      expectTypeOf(query1.data!).toMatchTypeOf<number[]>();
+
+      return (
+        <pre>
+          {query1.status}:{query1.isFetching ? 'fetching' : 'notFetching'}
+        </pre>
+      );
+    }
+
+    const utils = render(
+      <App>
+        <MyComponent />
+      </App>,
+    );
+    await waitFor(() => {
+      expect(utils.container).toHaveTextContent(`success:notFetching`);
+    });
+
+    expect(states).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "data": undefined,
+          "status": "pending",
+        },
+        Object {
+          "data": Array [],
+          "status": "pending",
+        },
+        Object {
+          "data": Array [
+            1,
+          ],
+          "status": "pending",
+        },
+        Object {
+          "data": Array [
+            1,
+            2,
+          ],
+          "status": "pending",
+        },
+        Object {
+          "data": Array [
+            1,
+            2,
+            3,
+          ],
+          "status": "success",
+        },
+      ]
+    `);
+    expect(states.map((s) => s.status)).toEqual([
+      // initial
+      'pending',
+      // waiting 3 values
+      'pending',
+      'pending',
+      'pending',
+      // done iterating
+      'success',
+    ]);
+  });
```

Comment:
My proposal for that test would be to also try observing the `fetching` state in addition to the `loading` state, as it could help inform the various possible `setState` used in the queryFn

```suggestion
  test('iterable', async () => {
    const { client, App } = ctx;
    const states: Partial<QueryState>[] = [];
    function MyComponent() {
      const query1 = client.post.iterable.useQuery(undefined, {
        trpc: {
          context: {
            stream: 1,
          },
        },
      });
      states.push({
        status: query1.status,
        data: query1.data,
        fetchStatus: query1.fetchStatus
      });

      expectTypeOf(query1.data!).toMatchTypeOf<number[]>();

      return (
        <pre>
          {query1.status}:{query1.isFetching ? 'fetching' : 'notFetching'}
        </pre>
      );
    }

    const utils = render(
      <App>
        <MyComponent />
      </App>,
    );
    await waitFor(() => {
      expect(utils.container).toHaveTextContent(`success:notFetching`);
    });

    expect(states).toMatchInlineSnapshot(`
      Array [
        Object {
          "data": undefined,
          "fetchStatus": "fetching",
          "status": "pending",
        },
        Object {
          "data": Array [],
          "fetchStatus": "fetching",
          "status": "pending",
        },
        Object {
          "data": Array [
            1,
          ],
          "fetchStatus": "fetching",
          "status": "pending",
        },
        Object {
          "data": Array [
            1,
            2,
          ],
          "fetchStatus": "fetching",
          "status": "pending",
        },
        Object {
          "data": Array [
            1,
            2,
            3,
          ],
          "fetchStatus": "idle",
          "status": "success",
        },
      ]
    `);
    expect(states.map((s) => `${s.status}-${s.fetchStatus}`)).toEqual([
      // initial
      'pending-fetching',
      // waiting 3 values
      'pending-fetching',
      'pending-fetching',
      'pending-fetching',
      // done iterating
      'success-idle',
    ]);
  });
```